cli/cmd/plugin/unmanaged-cluster/cluster/kind.go

```diff
@@ -198,6 +202,69 @@ func kindConfigFromClusterConfig(c *config.UnmanagedClusterConfig) ([]byte, erro
 	return rawConfig.Bytes(), nil
 }
 
+// mergeIntoProviderConfig merges the default kind config created via flags
+// into the kind config created via the provider config
+//
+// Provider configs take precidence.
+// So options given via `rawKindConfig` are the highest order configurations.
```

Comment:
I wonder if it would be better to give command line arguments priority over rawKindConfig. That would allow someone to have a common base kind config that they always use that has the defaults they usually want, but then adjust it at deploy time by passing different arguments to the create command.

Reply:
> That would allow someone to have a common base kind config that they always use that has the defaults they usually want, but then adjust it at deploy time by passing different arguments to the create command.


That's a cool workflow. I like that idea. But I honestly have no preference. 

It would probably be good to define this in writing soon so we can have an expected _ethos_ users can refer to when creating configs / scripts, and we can have something to fall back on when these discussions come up.

Something like:
```
1. Command line arguments take precedence over all
2. Env variables take precedence over config files
3. Configuration files take precedence over default settings
4. Default values are used if nothing is provided
```

---

There is this comment which should probably go in the docs?
https://github.com/vmware-tanzu/community-edition/blob/690018ecd05cdbb8071e6e388463f6604a0abd24/cli/cmd/plugin/unmanaged-cluster/config/config.go#L168-L176

Reply:
We do sort of have it documented here: https://tanzucommunityedition.io/docs/v0.11/ref-unmanaged-cluster/#custom-configuration

Not sure if someone would actually read that far, but since we do publicly state that, it would be good to stay consistent with that order of precedence.

Reply:
I agree 100% with @stmcginnis as this is typical overriding of configuration in many programs I know and use.